flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java

```diff
@@ -877,4 +899,610 @@ private String getSensorPositionInfo(Asset asset) {
+               serverVoltage = data.getValueNum();
+            }
+            if(outLet.equals(extraidentifier)) {
+               data.setMetricName(String.format(MetricKeyName.SERVER_CONNECTED_PDUX_OUTLETX_VOLTAGE, pduAssetId, value.getExtraidentifier()));
+               result.add(data);
+               serverVoltage = data.getValueNum();
+            }
+            break;
+         default:
+            break;
+         }
+      }
+      if(serverVoltage != null) {
+         MetricData data = new MetricData();
+         data.setMetricName(MetricName.SERVER_VOLTAGE);
+         data.setTimeStamp(serverVoltageReadTime);
+         data.setValueNum(serverVoltage);
+         data.setUnit(MetricUnit.V.toString());
+         result.add(data);
+      }
+      return result;
+   }
+
+   private List<MetricData> generateSensorMetricDataForServer(List<ValueUnit> valueUnits, String location, String metricName){
+      List<MetricData> result = new ArrayList<MetricData>();
+      if(valueUnits == null || valueUnits.isEmpty()) {
+         return result;
+      }
+      if(location.indexOf(FlowgateConstant.SEPARATOR) > -1) {
+         location = location.replace(FlowgateConstant.SEPARATOR, FlowgateConstant.UNDERLINE);
+      }
+      for(ValueUnit value : valueUnits) {
+         MetricData data = new MetricData();
+         data.setTimeStamp(value.getTime());
+         data.setValueNum(value.getValueNum());
+         String unit = databaseUnitAndOutputUnitMap.get(value.getUnit());
+         if(unit == null) {
+            unit = value.getUnit();
+         }
+         data.setUnit(unit);
+         switch (value.getKey()) {
+         case MetricName.HUMIDITY:
+            switch (metricName) {
+            case MetricName.SERVER_BACK_HUMIDITY:
+               data.setMetricName(String.format(MetricKeyName.SERVER_BACK_HUMIDITY_LOCATIONX, location));
+               result.add(data);
+               break;
+            case MetricName.SERVER_FRONT_HUMIDITY:
+               data.setMetricName(String.format(MetricKeyName.SERVER_FRONT_HUMIDITY_LOCATIONX, location));
+               result.add(data);
+               break;
+            default:
+               break;
+            }
+            break;
+         case MetricName.TEMPERATURE:
+            switch (metricName) {
+            case MetricName.SERVER_BACK_TEMPREATURE:
+               data.setMetricName(String.format(MetricKeyName.SERVER_BACK_TEMPREATURE_LOCATIONX, location));
+               result.add(data);
+               break;
+            case MetricName.SERVER_FRONT_TEMPERATURE:
+               data.setMetricName(String.format(MetricKeyName.SERVER_FRONT_TEMPERATURE_LOCATIONX, location));
+               result.add(data);
+               break;
+            default:
+               break;
+            }
+            break;
+         default:
+            break;
+         }
+      }
+      return result;
+   }
+
+   private Map<String, Map<String, String>> translateFormulaMap(Asset asset){
```

Comment:
Can you describe what input and what is the output?  what will this method used for?